src/MigrationDeriver.php

```diff
@@ -335,13 +335,16 @@ public function createAll(RequestInterface $request) {
    * {@inheritdoc}
    */
   public function deleteAll(RequestInterface $request) {
-    // Nuke the storage for the given mgiration group.
+    // Nuke the storage for the given migration group.
+    /** @var \Drupal\migrate_plus\Entity\MigrationInterface[] $migrations */
     $migrations = $this->migrationStorage->loadByProperties([
       'migration_group' => $this->migrationGroupDeriver->deriveName($request),
     ]);
 
     // Nuke the message and map tables for the given migrations.
-    foreach ($migrations as $migration) {
+    foreach ($migrations as $migration_entity) {
+      /** @var \Drupal\migrate\Plugin\MigrationInterface $migration */
+      $migration = $this->migrationPluginManager->createInstance($migration_entity->id());
       $migration->getIdMap()->destroy();
     }
```

Comment:
Not sure that this was ever working... long-standing issue where deleting requests would've likely been complaining about a non-existent `::getIdMap()` method?

That said: When the request is deleted, the ID map and message tables should be cleaned up, so we just need some kind of expiration method around things?

Reply:
Tested this, huzzah.